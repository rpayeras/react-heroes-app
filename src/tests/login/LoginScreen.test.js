import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import { LoginScreen } from "../../components/login/LoginScreen";
import { types } from "../../types/types";

const mockNavigate = jest.fn();

const context = {
  user: {
    logged: false,
  },
  dispatch: jest.fn(),
};

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("Testing LoginScreen", () => {
  const wrapper = mount(
    <AuthContext.Provider value={context}>
      <MemoryRouter initialEntries={["/login"]}>
        <LoginScreen />
      </MemoryRouter>
    </AuthContext.Provider>
  );

  test("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should do dispatch and navigation", () => {
    wrapper.find(".btn.btn-primary").simulate("click");

    expect(context.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: "User",
      },
    });

    expect(mockNavigate).toHaveBeenCalledWith("/marvel");

    localStorage.setItem("lastPath", "/dc");

    wrapper.find(".btn.btn-primary").simulate("click");

    expect(mockNavigate).toHaveBeenCalledWith(
      localStorage.getItem("lastPath") || "/marvel"
    );
  });
});
