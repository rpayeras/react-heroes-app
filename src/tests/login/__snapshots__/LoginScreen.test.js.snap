// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginScreen should render properly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
